misc/plot-bamstats

```diff
@@ -1204,6 +1207,119 @@ sub plot_indel_cycles
     plot($$args{gp});
 }
 
+sub plot_read_len_hist
+{
+    my ($opts) = @_;
+
+    my @rl = get_values($opts, 'RL');
+    if ( !@rl ) { return; }
+
+    # TODO: Add modified version for paired end reads.
+    my $is_paired = get_value($opts,'SN','reads paired:');
+
+    my $args = get_defaults($opts,"$$opts{prefix}read_length.png");
+    my $fh = $$args{fh};
+
+    # anonymous subroutines for code readibility
+    my $log_bin = sub {
+        my $x = $_[0];
+        return floor((log($x) / log(10)) * 10);
+    };
+    my $inv_log_bin = sub {
+        my $x = $_[0];
+        return floor(10**($x/10));
+    };
+
+    # Open files to store data: this is more elegant than copying four sets of 
+    # data into a single GNUplot file.
+    open(my $rd_fh, '>', "read_len_rawdata.txt");
+    open(my $hd_fh, '>', "read_len_histdata.txt");
+    
+    # Create gnuplot command file options
+    my $min_x = $rl[0][0];
+    my $max_x = $rl[-1][0];
+    # Gnuplot commands for normalized binned data
+    print $fh qq[
+        set y2tic
+        set terminal png size 800,1000 truecolor
+        set output "$$args{img}"
+        $$args{grid}
+        set multiplot layout 2,1 columnsfirst scale 1,0.9
+        set autoscale y
+        set ylabel "Read Count"
+        set autoscale x
+        set logscale x
+        set xrange[$min_x: $max_x]
+        set xtics rotate by -45 scale 0
+        set xlabel "Read Length"
+        set boxwidth .8 relative
+        set title "$$args{title}" noenhanced 
+        set title offset -15,-.5 
+        set key at graph 1, 1.15
+        plot "read_len_rawdata.txt" using 1:2 lt rgb "orange" with boxes fs solid .7 title 'unbinned', "read_len_histdata.txt" using 4:3 lt rgb "blue" with boxes title 'normalized bins(count/bin width)' axis x1y2
```

Comment:
It took a while before I understood why we had two plots, and I've read the discussion :-).  Then I remembered it's because we have the normalised-by-bin-size and unnormalised versions.  I see these in the key, so there is a hint if you look, but it may not be that obvious to many.  

Without the background knowledge I'd simply be asking myself "normalised how?"!
Sadly it appears plot-bamstats has no man page or documentation, so people have to figure it out.  This isn't your fault - it's just where we're starting from.

However perhaps we can get a comment in the source code so if people read the code to understand what these two plots are and what the thing we're normalising over is, then it'll help.

Reply:
I've added some explanation to both of the graphs to explain how they are derived, what they show, and the logic behind why they were included. The lack of documentation on plot-bamstats is disappointing. Do you think that writing some documentation for it would be useful?

Reply:
I went ahead and merged as-is.  Documentation would clearly be useful, but there is quite a lot that's missing from plot-bamstats and a considerable amount to document.  If you wished to write documentation for it then we'd be grateful, but it's different to this issue and PR and it shouldn't delay the merging of the new feature.  So if you wanted to do this, please let us know so we can discuss formats, and then create a new PR.  Totally understood if you have other things to do too.  (We do which is why it's never been done!)

Either way, thanks.